CloudSQLReplicator/tests/util/cloudsql_test.py

```diff
@@ -0,0 +1,322 @@
+                                          mock_auth_default):
+        """Test create_source_representation util function with defaults."""
+        del mock_auth_default  # unused in body of test
+        mock_build.return_value = MagicMock()
+        mock_insert_instance = (
+            mock_build.return_value.instances.return_value.insert
+        )
+        test_db_name = 'test-external-mysql-representation'
+        test_host_port = '192.161.0.1/4321'
+        test_body = {
+            'name': test_db_name,
+            'databaseVersion': cloudsql.DEFAULT_2ND_GEN_DB_VERSION,
+            'region': cloudsql.DEFAULT_2ND_GEN_REGION,
+            'onPremisesConfiguration': {
+                'kind': 'sql#onPremisesConfiguration',
+                'hostPort': test_host_port
+            }
+        }
+        cloudsql.create_source_representation(None,
+                                              None,
+                                              source_body=test_body)
+        # Verify that instances.insert is called with specified arguments
+        mock_insert_instance.assert_called_with(
+            project=TEST_PROJECT,
+            body=test_body
+        )
+
+    @mock.patch.object(
+        uuid, 'uuid4',
+        return_value=TEST_UUID
+    )
+    def test_create_replica_instance_defaults(self,
+                                              mock_uuid,
+                                              mock_build,
+                                              mock_auth_default):
+        """Test create_replica_instance util function with defaults."""
+        del mock_uuid, mock_auth_default  # unused in body of test
+        mock_build.return_value = MagicMock()
+        mock_insert_instance = (
+            mock_build.return_value.instances.return_value.insert
+        )
+
+        cloudsql.create_replica_instance(TEST_MASTER,
+                                         TEST_DUMPFILE_PATH,
+                                         TEST_REPL_USER,
+                                         TEST_PASSWORD)
+
+        # Verify that instances.insert is called with arguments and defaults
+        mock_insert_instance.assert_called_with(
+            project=TEST_PROJECT,
+            body={
+                'name': 'cloudsql-replica-{}'.format(TEST_UUID),
+                'settings': {
+                    'tier': cloudsql.DEFAULT_2ND_GEN_TIER,
+
+                },
+                'databaseVersion': cloudsql.DEFAULT_2ND_GEN_DB_VERSION,
+                'masterInstanceName': TEST_MASTER,
+                'region': cloudsql.DEFAULT_2ND_GEN_REGION,
+                'replicaConfiguration': {
+                    'mysqlReplicaConfiguration': {
+                        'dumpFilePath': TEST_DUMPFILE_PATH,
+                        'username': TEST_REPL_USER,
+                        'password': TEST_PASSWORD,
+                    }
+
+                }
+            }
+        )
+    def test_create_replica_instance(self,
```

Comment:
line spacing; you need a line space between this and the line above; i am surprised that the linter isn't catching this

Reply:
Done.